### ISSUE TITLE
`prefer_final_locals ` false positive for destructured tuple in loop.

### ISSUE THREAD

**Issue by cranst0n** [dart-lang/linter#4286]
**Describe the issue**
I'm seeing a lint warning about `Local variables should be final` when destructuring a list of tuples inside a loop. This example uses Dart 3.

If you're seeing unexpected behavior in a lint rule, please include the name of the rule in the issue description.
prefer_final_locals

**To Reproduce**
```dart
for (final (a, b) in [(1, 2), (3, 4), (5, 6)]) {
  print('A: $a   B: $b');
}
```

**Expected behavior**
Should not see a lint warning about about `Local variables should be final` for the `a` and `b` variables.

**Additional context**
n/a

**Comment by pq** [comment]
Confirmed. Looking now...

Thanks for the report!
